src/gtc/dace/transformations.py

```diff
@@ -0,0 +1,112 @@
+import dace
+from dace.transformation.dataflow import TrivialMapElimination
+from dace.transformation.helpers import nest_state_subgraph
+from dace.transformation.interstate import InlineTransients
+
+
+def eliminate_trivial_maps(sdfg: dace.SDFG):
+    """Remove maps and map ranges where the iteration is over a single index."""
+    applied = True
+    while applied:
+        applied = False
```

Comment:
When I first read this I thought it was a bug, but I see what you're doing here. Is there a way to write the termination condition in a way that it doesn't look like one?

Reply:
The problem is the double loop, so a `break` won't help us Would you like this better?
```
def _apply_once()
    for map_entry, state in filter(_filter_map_entries, sdfg.all_nodes_recursive()):
        try:
            TrivialMapElimination.apply_to(
                state.parent, map_entry=map_entry, verify=True, save=False
            )
            return True
        except ValueError:
          continue
    return False
    
while _apply_once()
    pass
```

Reply:
Never mind, I redid the entire thing using dace transformation API, not sure why I thought this wasn't possible before.